rasa/core/training/generator.py

```diff
@@ -133,12 +134,24 @@ def copy(
 
         return tracker  # yields the final state
 
+    def _freeze_current_state(self, state) -> frozenset:
```

Comment:
I think you use this function in featurizers as well, let's move it into utils as global faunction

Reply:
or maybe the function of the domain or tracker, not sure which one is better. For both `freeze` and `unfreeze` states